### PR TITLE
API Listing UI fixes #2498

### PR DESCRIPTION
**Pull request by Minoli**
### Proposed changes in this pull request

-Fixing https://github.com/wso2/product-apim/issues/2498

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/Listing.js [hunk 1]
```diff
@@ -82,47 +82,51 @@ class Listing extends React.Component {
...
+                                                        <TableRow>
+                                                            <TableCell>Name</TableCell>
+                                                            <TableCell>Version</TableCell>
+                                                            <TableCell>Context</TableCell>
+                                                            <TableCell>Description</TableCell>
+                                                        </TableRow>
+                                                    </TableHead>
+                                                    <TableBody>
+                                                        {this.state.apis.list.map(api => {
+                                                            return <APiTableRow api={api} key={api.id}/>
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/Listing.js]: I think we can simplify this as 

```javascript
this.state.apis.list.map( api => <APiTableRow api={api} key={api.id} )
```

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/Listing.js [hunk 2]
```diff
@@ -82,47 +82,51 @@ class Listing extends React.Component {
...
+                        <BottomNavigation value={value} onChange={this.handleChange} style={{float: "right"}}>
+                            <BottomNavigationButton icon={<ListIcon/>}/>
+                            <BottomNavigationButton icon={<GridOnIcon/>}/>
+                        </BottomNavigation>
+                        {
+                            this.state.apis ?
+                                this.state.listType === "grid" ?
+                                    <Grid container >
+                                        {this.state.apis.list.map(api => {
+                                            return <ApiThumb api={api} key={api.id}/>
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/Listing.js]: Same as below, if arrow function has only one statement no need to put curly braces and return, instead simply put the single statement inline

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/ApiThumb.js [hunk 3]
```diff
@@ -82,45 +82,18 @@ class ApiThumb extends React.Component {
...
-                                //todo:Delete button should be enabled after M6 based on permission model
-                            </DialogActions>
-                        </Dialog>
-                    </CardActions>
-                </Card>
+            <Grid item xs={12} sm={6} md={3} lg={3} xl={3}>
+                <Link to={details_link}>
+                    <GridListTile className="api-list-tile">
+                        <Avatar className="default-thumb" style={{backgroundColor: randomApiThumbColor}}>
+                                    {firstTwoLetters(name)}
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/ApiThumb.js]: shall we move the method call out of jsx context?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Base/Header/Header.js [hunk 4]
```diff
@@ -205,8 +205,8 @@ class Header extends React.Component {
                                 {/* User menu */}
                                 <Button aria-owns="simple-menu" aria-haspopup="true" onClick={this.handleClickUserMenu}
                                         color="contrast">
-                                    <Avatar alt="{user.name}" src="/store/public/app/images/users/user.png"></Avatar>
-                                    <span>{user.name}</span>
+                                    <Avatar alt="{user.name}" src="/store/public/app/images/users/user.png" className="header-avatar"></Avatar>
```
Comment (jeradrutnam) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Base/Header/Header.js]: Fix character limit.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/App.css [hunk 5]
```diff
@@ -69,3 +69,29 @@ header.MuiAppBar-colorPrimary-6{
     font-size:15px;
     font-weight: 200;
 }
+.parent-container {
+    padding: 82px 0;
+}
+.add-item-left-padding{
+    padding-left: 30px;
+}
```
Comment (jeradrutnam) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/App.css]: Have a space before `{`

**Pull request merged**